divide_and_conquer/mergesort.py

```diff
@@ -27,13 +27,16 @@ def merge(arr, left, mid, right):
     return arr
 
 
-def mergesort(arr, left, right):
+def mergesort(arr, left=None, right=None):
     """
-    >>> mergesort([3, 2, 1], 0, 2)
+    >>> mergesort([3, 2, 1])
     [1, 2, 3]
-    >>> mergesort([3, 2, 1, 0, 1, 2, 3, 5, 4], 0, 8)
+    >>> mergesort([3, 2, 1, 0, 1, 2, 3, 5, 4])
```

Comment:
Please add a doctest for `mergesort([3, 2, 1], -1, -3.1)`

Reply:
Added that doctest and also `mergesort([3, 2, 1], None, False)`